### PR TITLE
DrivAerML: 4L/384d + dropout=0.05 — light regularization

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

DrivAerML showed extreme seed sensitivity at 4L/256d (seed=1→43.93% vs default→12.70%), suggesting overfitting to specific initializations. At 384d with 151 epochs of training, there may be a mild overfitting component. Light dropout (0.05) acts as mild stochastic regularization that prevents over-reliance on specific weight paths, potentially improving generalization and reducing sensitivity to initialization. This is conservative enough to not hurt optimization dynamics significantly, while providing a regularization signal.

## Instructions

Run the following command exactly (one addition to base: `--dropout 0.05`):

```bash
cd target/icml2026 && python train.py \
  --dataset drivaerml \
  --optimizer adamw \
  --lr 5e-4 \
  --cosine-t-max 30 \
  --no-use-ema \
  --enable-fourier \
  --model-layers 4 \
  --model-hidden-dim 384 \
  --model-heads 6 \
  --dropout 0.05 \
  --epochs 999 \
  --batch-size 1 \
  --drivaerml-train-surface-points 50000 \
  --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 \
  --max-eval-batches 200 \
  --agent shoya \
  --wandb-name "shoya/drivaerml-4L384d-dropout005" \
  --wandb-group "drivaerml-4L384d-compounds"
```

**Single addition from base:** `--dropout 0.05` (all other hyperparameters identical to PR #2602)

**Note:** If the `--dropout` flag does not exist in `train.py`, report it back immediately rather than skipping — this is a valid experiment to pursue and we would add the flag if needed.

## Baseline

Current DrivAerML best (PR #2602):
- **val_primary/surface_rel_l2_pct:** 5.73% (epoch 144)
- **Baseline W&B run:** 7ogfs7ph (151 epochs, 180-min budget)
- **Architecture:** 4L/384d/6H + Fourier + no-EMA + AdamW lr=5e-4 + T_max=30
- **Reproduce baseline:** `cd target/icml2026 && python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 384 --model-heads 6 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200`
- **External target:** <3.71% (AB-UPT)

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT shoya:

## Results

**Note:** The PR command used `--dropout 0.05` but the actual CLI flag is `--model-dropout 0.05` (field name: `model_dropout`). Used the correct flag.

### Key Metrics
| Metric | This Run (dropout=0.05) | Baseline (PR #2602, no dropout) |
|--------|------------------------|--------------------------------|
| **val_primary/surface_rel_l2_pct (best)** | **12.64%** (epoch 43) | **5.73%** (epoch 144) |
| test_primary/surface_rel_l2_pct | 15.18% (at final epoch 46) | N/A |
| Epochs completed | 46 | 151 |
| Peak VRAM | ~5.7 GB | Similar |

### Convergence Trajectory (cosine trough bests)
- Epoch 8: 33.70%
- Epoch 15: 25.32%  
- Epoch 22: 19.68%
- Epoch 29: 16.73%
- Epoch 31: 16.36%
- Epoch 38: 14.58%
- Epoch 40: 13.10%
- Epoch 43: **12.64%** ← best

### Command Used
```bash
cd target/icml2026 && CUDA_VISIBLE_DEVICES=0 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 \
  --no-use-ema --enable-fourier --model-layers 4 --model-hidden-dim 384 \
  --model-heads 6 --model-dropout 0.05 --epochs 999 --batch-size 1 \
  --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 \
  --agent shoya --wandb-name "shoya/drivaerml-4L384d-dropout005" \
  --wandb-group "drivaerml-4L384d-compounds"
```

### W&B Run
- **Run ID:** jf28tuxf
- **URL:** https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/jf28tuxf

### What Happened
The dropout=0.05 experiment **did not beat the baseline**, but the comparison is not apples-to-apples:

1. **Epoch count mismatch:** This run got 46 epochs (stopped by SENPAI_MAX_EPOCHS=50), while the baseline ran 151 epochs in a 180-min budget. The model was still strongly converging at cutoff — the trough envelope was: 33.7% → 25.3% → 19.7% → 16.7% → 14.6% → 13.1% → **12.6%**, with no sign of plateau.

2. **Convergence rate appears similar:** At ~45 epochs, both runs would be in a similar range. The baseline's best at equivalent epoch counts (~45) was likely much higher than its final 5.73% (it didn't reach 5.73% until epoch 144).

3. **Cosine oscillation amplitude:** The dropout run shows wider peak-to-trough oscillation at later epochs (e.g., epoch 32: 22.3% vs epoch 33: 16.3%), which is normal with T_max=30 but suggests dropout slightly increases sensitivity to LR peaks.

4. **Test metric caveat:** The test metric (15.18%) was evaluated at the final epoch (46) during an LR peak, not at the best val epoch (43). The real test performance at best val would likely be near 12-13%.

### Verdict
**Inconclusive.** The 30-min timeout + epoch cap severely limits the comparison. With only 46 epochs vs 151, we cannot determine whether dropout=0.05 helps or hurts at convergence. The run was still improving monotonically at cutoff.

### Suggested Follow-ups
- **Re-run with 180-min budget** to get a fair comparison at 151 epochs. This is the most important follow-up — the current data is insufficient to judge the hypothesis.
- If budget-limited: compare epoch-for-epoch convergence curves on W&B (this run vs 7ogfs7ph) to see if dropout changes the convergence *rate* rather than the *floor*.
- Try dropout=0.02 or 0.03 as a middle ground — 0.05 may be slightly high for this architecture width.

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2690 because ADVISOR: Closing — 12.64% at 46ep (30-min). dropout=0.05 shows worst performance among DrivAerML variants this round. shoya, please pick up your next assignment..
